### PR TITLE
EternalNetwork TM Presence

### PR DESCRIPTION
**Pull request by TheDjRider**
### Tracking for using websites of EternalNetwork TM.
With settings for what part to show and **_multilanguage_** support.

In the next updates probably will be adding buttons to open currently viewed pages and more custom messages for specific places/actions.

Main

Avaiable translaions & changed main logo:
Main

Main

### **Website:**
Website

Website

### **Forum:**
Forum

Forum

Forum

### **Radio**
Radio

### **Status:**
Status

### CODE REVIEW

File: websites/E/EternalNetwork TM/presence.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    if (!showSubdomain) {
+      presenceData.smallImageKey = undefined;
+      presenceData.smallImageText = undefined;
+    }
+
+    if (presenceData.details == null) {
+      presence.setTrayTitle();
+      presence.setActivity();
+    } else 
+      presence.setActivity(presenceData);
```
Comment (EncryptedDev) [inline websites/E/EternalNetwork TM/presence.ts:148]:
```suggestion
      presence.setActivity(presenceData);
```

File: websites/E/EternalNetwork TM/presence.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
```
Comment (EncryptedDev) [inline websites/E/EternalNetwork TM/presence.ts:149]:
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/E/EternalNetwork TM/presence.ts [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        presenceData.state = "Adding new service !";
+    }
+
+    if (!showSubdomain) {
+      delete presenceData.smallImageKey;
+      delete presenceData.smallImageText;
+    }
+
+    presence.setActivity(presenceData);
+
```
Comment (EncryptedDev) [inline websites/E/EternalNetwork TM/presence.ts:127]:
```suggestion
```

**Review by Bas950 — CHANGES_REQUESTED** [review]
Fix errors.

File: websites/E/EternalNetwork TM/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  "$schema": "https://schemas.premid.app/metadata/1.4",
+  "service": "EternalNetwork TM",
+  "author": {
+    "name": "TheDjRider",
+    "id": "149202464590004224"
+  },
+  "category": "other",
+  "logo": "https://i.imgur.com/Saq6ADh.png",
+  "thumbnail": "https://i.imgur.com/8pBRB2e.png",
+  "version": "1.0.1",
```
Comment (Bas950) [inline websites/E/EternalNetwork TM/dist/metadata.json:11]:
```suggestion
  "version": "1.0.0",
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/E/EternalNetwork TM/dist/metadata.json (hunk 4 above)
Comment (TheDjRider) [reply websites/E/EternalNetwork TM/dist/metadata.json]: Done.

**Review by TheDjRider — COMMENTED** [review]
Changes are applied

File: websites/E/EternalNetwork TM/dist/metadata.json [hunk 5]
```diff
@@ -0,0 +1,62 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.4",
+  "service": "EternalNetworkTM",
```
Comment (Bas950) [inline websites/E/EternalNetwork TM/dist/metadata.json:3]:
```suggestion
  "service": "EternalNetwork TM",
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Comment by TheDjRider** [comment]
@Bas950 so what should be the name in **metadata.json** when I check other Presence it has many different variations. 

Mine was **_EternalNetwork TM_**, and the verifier found an error, now is **_EternalNetworkTM_** (without space) and has a problem again.

For example: "_Edge Me Please_" is not a problem and it has spaces and bug letters. 
Where is the problem with mine?

File: websites/E/EternalNetwork TM/dist/metadata.json (hunk 5 above)
Comment (TheDjRider) [reply websites/E/EternalNetwork TM/dist/metadata.json]: Done

**Review by Slowlife01 — DISMISSED** [review]

**Review by Bas950 — APPROVED** [review]

File: websites/E/EternalNetwork TM/dist/metadata.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  "thumbnail": "https://i.imgur.com/8pBRB2e.png",
+  "version": "1.0.0",
+  "color": "#c005e6",
+  "tags": [
+    "Community",
+    "Network",
+    "Gaming",
+    "News",
+    "Radio",
+    "Forum"
```
Comment (Bas950) [inline websites/E/EternalNetwork TM/dist/metadata.json:19]:
```suggestion
    "community",
    "network",
    "gaming",
    "news",
    "radio",
    "forum"
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by Slowlife01 — APPROVED** [review]

**Pull request merged**